### DISCUSSION TITLE
Motor Encoders didn't work

### DISCUSSION THREAD

**Discussion by AdibSuid** [hippo5329/linorobot2_hardware#16] [Q&A]
I successfully uploaded firmware into teensy40 and the map was generated in the rviz. When I rotate the robot, the map updated but When I want to move the robot linearly x or y, the robot doesn't update it's position. The teleop twist keyboard also didn't work if I connect all the encoders. It works fine when I manually move the robot using teleop twist keyboard without encoder pins connected but once I connected them, when I run micro ros agent or robot bringup, the motors vibrates like crazy, counter clockwise and clockwise until I push the emergency stop button.

**Comment by hippo5329** [comment]
Sorry, I missed this message. Please run test_motors to verify motors and encoders.

https://github.com/hippo5329/linorobot2_hardware/wiki#test-the-motors-and-encoders

**Comment by hippo5329** [comment]
The io pins of 32 bits micro-controllers, ash as teensy pico esp32, are 3.3V tolerant. You should power the encoders with 3.3V.
